webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/VaultPath.scala

```diff
@@ -1,15 +1,49 @@
 package com.scalableminds.webknossos.datastore.datavault
 
+import com.aayushatharva.brotli4j.Brotli4jLoader
+import com.aayushatharva.brotli4j.decoder.BrotliInputStream
 import com.scalableminds.util.io.ZipIO
 import net.liftweb.util.Helpers.tryo
 
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream, IOException}
 import java.net.URI
 import scala.collection.immutable.NumericRange
 
 class VaultPath(uri: URI, dataVault: DataVault) {
 
-  def readBytes(range: Option[NumericRange[Long]] = None): Option[Array[Byte]] =
-    tryo(dataVault.readBytes(this, range)).toOption.map(ZipIO.tryGunzip)
+  def readBytes(range: Option[NumericRange[Long]] = None): Option[Array[Byte]] = {
+    val resultOpt = tryo(dataVault.readBytes(this, range)).toOption
+    resultOpt match {
+      case Some((bytes, encoding)) =>
+        encoding match {
+          case Encoding.gzip   => Some(ZipIO.tryGunzip(bytes))
```

Comment:
```suggestion
          case Encoding.gzip   => Some(ZipIO.gunzip(bytes))
```
(I think the tryGunzip method was only used here, can be removed now)

Reply:
Actually also used in FileSystemVaultPath. I guess it makes sense there because we have no further information on encoding?

Reply:
I think on file system we should not expect compression (beyond that which is specified in the zarr header and its equivalents, but that is already taken care of elsewhere)